Shopify/cli
# Simple no color

### WHY are these changes introduced?

Closes https://github.com/Shopify/internal-cli-foundations/issues/20
Closes https://github.com/Shopify/cli/issues/1161

We want to allow the user to disable color if so desired. This might apply for cloud environments, colorblind users, etc.



### WHAT is this pull request doing?



Utilizes the fact that `chalk` respects `FORCE_COLOR=0` to disable terminal colors. We detect a few different ENV variables based on the [CLIG guidelines](https://clig.dev/#output) and [no color standard](https://no-color.org/) and set `FORCE_COLOR=0` if we think the user has specified that they don't want colors.

Before
<img width="1491" alt="Screenshot 2023-04-02 at 15 45 56" src="https://user-images.githubusercontent.com/6288426/229353655-c02759c1-e884-4727-802d-116f4c3abac1.png">

After
<img width="1493" alt="Screenshot 2023-04-02 at 15 46 44" src="https://user-images.githubusercontent.com/6288426/229353692-dafa276e-d5d4-4d85-96a2-fda295f939cf.png">

### How to test your changes?

Run a command that uses colors, then specify any of:

* `NO_COLOR=1`
* `FORCE_COLOR=0`
* `SHOPIFY_FLAG_NO_COLOR=1`
* `TERM=dumb`
* `--no-color` flag

All should disable colors. The notable exception is themes, wherever we are handing off to the Ruby process, because assumptions about color are baked deeply into the underlying implementation of the Ruby output tools, and it will take significantly more time to handle that properly.

### Post-release steps



Add documentation

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
